tools/slang-capability-generator/capability-generator-main.cpp

```diff
@@ -657,22 +657,35 @@ void calcCanonicalRepresentations(DiagnosticSink* sink, List<RefPtr<CapabilityDe
         calcCanonicalRepresentation(sink, def, mapEnumValueToDef);
 }
 
-void outputUIntSetAsBufferValues(const String& nameOfBuffer, StringBuilder& resultBuilder, UIntSet& set)
+// Create a local UIntSet with data
+void outputLocalUIntSetBuffer(const String& nameOfBuffer, StringBuilder& resultBuilder, UIntSet& set)
 {
-    // store UIntSet::Element as uint8_t to stay sizeof(UIntSet::Element) independent.
-    // underlying type may change, bits stay the same.
-    resultBuilder << "inline static CapabilityAtomSet generate_" << nameOfBuffer << "()\n";
-    resultBuilder << "{\n";
-    resultBuilder << "    CapabilityAtomSet generatedSet;\n";
-
+    resultBuilder << "    CapabilityAtomSet " << nameOfBuffer << ";\n";
     for (Index i = 0; i < set.getBuffer().getCount(); i++)
     {
-        resultBuilder << "    generatedSet.addRawElement(UIntSet::Element(" << set.getBuffer()[i] << "), " << i << ");\n";
+        resultBuilder << "    " << nameOfBuffer << ".addRawElement(UIntSet::Element(" << set.getBuffer()[i] << "), " << i << "); \n";
```

Comment:
Can we reserve the buffer space before adding the elements? Something like "nameOfBuffer.reserveRawBuffer(set.getBuffer().getCount())`

Reply:
we can 👍